stone_age/game_board/get_something_throw.py

```diff
@@ -1,14 +1,20 @@
-from typing import List
-from stone_age.simple_types import Effect
+from __future__ import annotations
+from typing import Iterable
+from stone_age.simple_types import Effect, ActionResult
+from stone_age.game_board.simple_types import Player
+from stone_age.game_board.interfaces import EvaluateCivilizationCardImmediateEffect
+from stone_age.game_board.interfaces import InterfaceCurrentThrow
 
 
-class GetSomethingThrow:
-    def __init__(self, resource: List[Effect]):
-        self._resource = resource
+class GetSomethingThrow(EvaluateCivilizationCardImmediateEffect):
+    def __init__(self, current_throw: InterfaceCurrentThrow) -> None:
+        self._current_throw = current_throw
 
-    @property
-    def resource(self) -> List[Effect]:
-        return self._resource
+    def perform_effect(self, player: Player, choice: Iterable[Effect]) -> ActionResult:
+        try:
```

Comment:
I should have mentioned this earlier, this class needs to be reworked a bit. Player does not choose which resource will he throw for, this is given by card (the responsibility of `CivilisationCardPlace` class to call the correct `GetSomethingThrow` instance). Choice argument can be made empty, effect for which this class throws can be given by constructor (one instance per each effect).